staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -297,6 +316,12 @@ func (g *Cloud) ensureInternalLoadBalancerDeleted(clusterName, clusterID string,
 	if err := g.ensureInternalInstanceGroupsDeleted(igName); err != nil && !isInUsedByError(err) {
 		return err
 	}
+	if g.client != nil {
```

Comment:
same down here, mock client in the unit tests to avoid introducing branches for testing sake

Reply:
ok. I had started on the approach to use the mock client. We will also need to change all tests to also create the service so that the client.Get() will return a non-nil value. I had held off on it since it will be a much longer diff. I can include it with this PR.